### PR TITLE
Repair & medical object attributes for eden editor

### PR DESCRIPTION
**Pull request by SzwedzikPL**
http://i.imgur.com/LXfNjDB.jpg //units
http://i.imgur.com/ad0qBLq.jpg //vehicles
http://i.imgur.com/j6S7Tmo.jpg //props

### CODE REVIEW

**Comment by jokoho48** [comment]
:+1:

**Comment by jonpas** [comment]
Should we use a tag in those settings to set them apart from vanilla ones and make it clear it's an ACE setting? Eg. `[ACE] Is Medic`.

File: addons/medical/stringtable.xml [hunk 1]
```diff
@@ -3891,5 +3891,13 @@
             <Portuguese>Não existe nenhum torniquete nesta parte do corpo!</Portuguese>
             <Czech>Žádné škrtidlo na této části těla!</Czech>
         </Key>
+		<Key ID="STR_ACE_Medical_Attributes_isMedic_Description">
+			<English>Medical training</English>
+			<Polish>Wyszkolenie medyczne</Polish>
+		</Key>
+        <Key ID="STR_ACE_Medical_Attributes_isMedicalVehicle_Description">
+            <English>Whatever or not the objects will be a medical vehicle.</English>
```
Comment (jonpas) [inline addons/medical/stringtable.xml]: Probably meant to be `Whether` and `object`.

**Comment by Daishiknyte** [comment]
Jonpas is correct.  Any added commands, menus, etc., should clearly show the mod they're associated with.

**Comment by esteldunedain** [comment]
:+1: Nice. I agree that some indication that those settings are ace would be great. Maybe:
`ACE: Is Medic:`

**Comment by SilentSpike** [comment]
I feel like having categories for ace modules would make more sense than using the special states category. Also eliminates the need to repetitively label each attribute with ace in the name.

**Comment by commy2** [comment]
^ this

File: addons/medical/CfgEden.hpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
+class Cfg3DEN {
+    class Object {
+        class AttributeCategories {
+            class StateSpecial {
+                class Attributes {
+                    class ACE_IsMedic {
```
Comment (commy2) [inline addons/medical/CfgEden.hpp]: lower case is

File: addons/repair/CfgEden.hpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
+class Cfg3DEN {
+    class Object {
+        class AttributeCategories {
+            class StateSpecial {
+                class Attributes {
+                    class ACE_IsEngineer {
```
Comment (commy2) [inline addons/repair/CfgEden.hpp]: lower case is

**Comment by SzwedzikPL** [comment]
I've added ace category and used common_Options stringtable

http://i.imgur.com/Z1tI6b3.jpg //units
http://i.imgur.com/KRwpL0Y.jpg //vehicles
http://i.imgur.com/CNoeQJe.jpg //props

**Comment by gienkov** [comment]
:+1:
10.02.2016 01:18 "SzwedzikPL" <notifications@github.com> napisał(a):

**Comment by esteldunedain** [comment]
Looks good @SzwedzikPL

**Comment by esteldunedain** [comment]
:+1: For merging this for 3.5.0

**Comment by PabstMirror** [comment]
@SzwedzikPL 
These do not work in MP, only the server has the correct settings.
```
        class AttributesBase {
            class Default {
                expression = "_this setVariable ['%%s',_value,true];";
            };
```
The default uses the broadcast flag whereas these do not.
We might want to just use inheritance instead of redefining `expression`.
As far as I can tell only the server executes the code.

Also see notes in #3339

**Pull request merged**